### PR TITLE
fix: Colors can be stringified with expontential alphas, but cannot b…

### PR DESCRIPTION
**Pull request by babycannotsay**
[bug source](https://github.com/Qix-/color/issues/156)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -136,8 +136,11 @@ cs.get.hsl = function (string) {
 	if (!string) {
 		return null;
 	}
-
-	var hsl = /^hsla?\(\s*([+-]?(?:\d{0,3}\.)?\d+)(?:deg)?\s*,?\s*([+-]?[\d\.]+)%\s*,?\s*([+-]?[\d\.]+)%\s*(?:[,|\/]\s*([+-]?[\d\.]+)\s*)?\)$/;
+	/**
+	 * scientific-notation regexp
+	 * [+-]?(?=\.\d|\d)(?:0|[1-9]\d*)?(?:\.\d*)?(?:[eE][+-]?\d+)?
+	 */
```
Comment (Qix-) [inline index.js:142]: This comment is unnecessary.
```suggestion
```

**Review by Qix- — CHANGES_REQUESTED** [review]
One nit, and can you add a number of tests under the "invalid" section(s) that test a few invalid inputs? Such as `1e`, `e`, `0e-`, etc.

**Comment by babycannotsay** [comment]
There's a compatibility issue.
Before hsla(0, 0%, 0%, +.) is same as hsla(0, 0%, 0%, 1)
but now hsla(0, 0%, 0%, +.) will throw error.

**Comment by Qix-** [comment]
That's fine, `+.` is not valid input. Anyone relying on that is masking a bug.

**Review by Qix- — CHANGES_REQUESTED** [review]
Can you add tests for decimal notations as well? E.g. `127.88e4`, `0.2e3`, `.1e-4`, etc. Right now only `1e..` is being tested.

**Comment by babycannotsay** [comment]
I wanted to save the scientific notation regular expression in a variable (and then write a test case), and pass it to new Regexp to concatenate it with the previous regular expression, but it was a lot of escaping/and made the regular expression even harder to understand.

**Review by Qix- — CHANGES_REQUESTED** [review]
Sorry to keep nitting but also add them for `hsl` too please since they're two separate regexes.

**Comment by Qix-** [comment]
Sorry, my food arrived right as I published and I got sidetracked. Quesadillas take priority in my life.

Thanks for the PR, published as `1.9.0`!

**Pull request merged**